aiden-recommender/aiden_recommender/scrapers/utils.py

```diff
@@ -1,130 +1,18 @@
-            "profile.default_content_setting_values.geolocation": 2,
-            "profile.default_content_setting_values.automatic_downloads": 1,
-            "download.default_directory": r"C:\temp",
-            "download.prompt_for_download": False,
-            "download.directory_upgrade": True,
-            "safebrowsing.enabled": False,
-        },
-    )
-    return options
-
-
-class ChromeDriver:
-    def __init__(self):
-        self.options = get_chrome_options()
-        self.drivers = []
-        self.max_drivers = 5
-        self.idle_timeout = 30
-        self.lock = threading.Lock()
-
-    def close_idle_drivers(self):
-        while True:
-            time.sleep(self.idle_timeout)
-            with self.lock:
-                for driver in self.drivers[:]:
-                    if not getattr(driver, "current_task", None):
-                        driver.quit()
-                        self.drivers.remove(driver)
-
-    def start(self):
-        threading.Thread(target=self.close_idle_drivers, daemon=True).start()
-
-    def get_available_driver(self):
-        for _ in range(10):
-            with self.lock:
-                for driver in self.drivers:
-                    if not getattr(driver, "current_task", None):
-                        return driver
-                if len(self.drivers) < self.max_drivers:
-                    driver = webdriver.Chrome(options=self.options)
-                    self.drivers.append(driver)
-                    return driver
-                time.sleep(0.5)
-        raise RuntimeError("All drivers are busy")
-
-    def get(self, url):
-        driver = self.get_available_driver()
-        driver.current_task = threading.current_thread()
-        try:
-            driver.get(url)
-            html = self.wait_for_page_load(driver)
-            return BeautifulSoup(html, "html.parser")
-        finally:
-            driver.current_task = None
-
-    @staticmethod
-    def wait_for_page_load(driver):
-        while True:
-            state = driver.execute_script("return document.readyState")
-            if state == "complete":
-                return driver.page_source
-            time.sleep(1)
-
-    def fetch_pages(self, urls):
-        def _fetch_page(url):
-            return self.get(url)
-
-        threads = []
-        results = [None] * len(urls)
-
-        for i, url in enumerate(urls):
-            thread = threading.Thread(target=lambda idx, u: results.__setitem__(idx, _fetch_page(u)), args=(i, url))
-            threads.append(thread)
-            thread.start()
-
-        for thread in threads:
-            thread.join()
-
-        return results
-
-    def fetch_page(self, url):
-        return self.get(url)
-
-
-chrome_driver = ChromeDriver()
-chrome_driver.start()
-
-
-def cache(retention_period: timedelta, model: Type[T], source: str):
+def cache(retention_period: timedelta, model: Optional[Type[T]] = None, source: Optional[str] = "default"):
```

Comment:
It cannot work if model=None since we use the function `model.model_dump_json` and `model.validate`